### PR TITLE
Fix pending auto-update installing a stale build

### LINKED ISSUE #8284: After auto-update, Positron immediately has another update
This is most noticeable when on the daily updates because of the frequency of new builds.

Once Positron detects that an update is available, it won't check for another update. Unless Positron is killed (to avoid installing the update), this version is the one that will be installed.

The problem appears when not updating the same day the update becomes available. When delaying the update, another version is published but Positron won't check for it. The result is updating to the first update and, upon restart, it runs a check and wants to update to the latest.

## Steps to reproduce
1. Positron found an update (let's call this `update-1`)
2. Leave Positron open until another update is published (`update-2`)
3. Run 'Restart to Update'

When Positron has finished updating, it has updated to `update-1`. Shortly after startup, Positron checks for updates and finds `update-2`.

### PR DESCRIPTION
**Pull request by timtmok**
Fixes #8284

### Summary

When Positron finds a pending update and the user delays "Restart to Update," a second update published in the meantime was never picked up: the pending update just stayed pending until restart, at which point it installed the stale build and immediately found a newer one. This PR makes the pending-update path re-check the feed instead of assuming what it staged is still current:

- The overwrite check (`checkForOverwriteUpdates`) now runs on a recurring interval while `Ready`, on explicit "Check for Updates" (also allowed from the `Ready` state, not just `Idle`), and again right before `quitAndInstall` -- so a restart can't install a build the feed has already superseded.
- Comparisons against the feed disable Chromium's HTTP cache (`disableCache: true`), which was letting a stale feed response look authoritative.
- On a metered connection, the overwrite check is deferred (not silently dropped) and re-attempted once the connection is no longer metered.
- Win32's `supportsUpdateOverwrite` is re-enabled (disabled in #16083 as a stopgap, since the overwrite round used to strand a pending update by clearing it without a way back); a new `restorePendingUpdate()` puts the original pending update back whenever the overwrite round doesn't produce a newer one.
- Adds dev-only tooling to exercise the whole flow without waiting for two real builds to ship: a mock update feed server (`npm run mock-update-server`, or the **Mock Update Server** task), a `devUpdateTesting` mode that simulates the download/install states on an unsigned source build, and a **Developer: Stage Pending Update from Feed URL...** command to stage an arbitrary hosted build as pending. See `scripts/mock-update-server/README.md` for the full walkthrough.

#### New Features
- N/A

#### Bug Fixes
- Fix Positron installing a stale update after a newer one was published while a "Restart to Update" was pending (#8284)

### Validation Steps

Signed builds of this branch, for testing the real download/install flow rather than `devUpdateTesting`'s simulation:
- https://github.com/posit-dev/positron-builds/actions/runs/35229762744
- https://github.com/posit-dev/positron-builds/actions/runs/35232431766

@:update

Unit coverage: `positronQuitAndInstall.vitest.ts` and `positronIsLatestVersion.vitest.ts` exercise the overwrite check, the metered-connection deferral, and `quitAndInstall`'s pre-restart re-check.

Manual, against the mock feed (no signed build needed):

1. Add to a gitignored `product.overrides.json` at the repo root:
   ```json
   { "quality": "dev", "updateUrl": "http://localhost:8902/positron" }
   ```
2. Run the **Mock Update Server** task (or `npm run mock-update-server`), then launch Positron with `./scripts/code.sh`. Within 30s the tooltip reads **Restart to Update**.
3. Publish a newer version while Positron is still running:
   ```bash
   curl localhost:8902/publish/2026.09.0-3/bbbbbbb
   ```
4. Either wait up to 30s or click **Restart to Update** -- it should re-check first, pass through **Downloading Newer Update**, and settle back on **Restart to Update** with the new version instead of installing the stale one.
5. Click **Restart to Update** again; the main log should record it would install the newer version.

On macOS/Windows with a signed build, **Developer: Stage Pending Update from Feed URL...** can stage an older hosted build as pending and then verify the real overwrite/download round completes against the actual channel feed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/vs/platform/update/electron-main/abstractUpdateService.ts [hunk 1]
```diff
@@ -634,6 +729,16 @@ export abstract class AbstractUpdateService extends Disposable implements IUpdat
 			return Promise.resolve(undefined);
 		}
 
+		if (this.supportsUpdateOverwrite && !this._hasCheckedForOverwriteOnQuit) {
+			this._hasCheckedForOverwriteOnQuit = true;
```
Comment (jonvanausdeln) [inline src/vs/platform/update/electron-main/abstractUpdateService.ts:733]: `_hasCheckedForOverwriteOnQuit` is set here and never cleared, so the pre-restart re-check is once per *process* rather than once per restart attempt. That is the right shape for the loop it guards against, but it leaves a hole on the veto path:

1. Build A is pending; user clicks **Restart to Update**.
2. The flag flips to `true`, the re-check finds nothing newer, and `lifecycleMainService.quit(true)` is vetoed (unsaved editor, user cancels the dialog).
3. The veto handler restores `Ready`, and `setState()` re-arms `overwriteUpdatesCheckInterval` for another 5 minutes.
4. Build B publishes inside that window and the user clicks **Restart to Update** again. The flag is still `true`, so no re-check runs and stale build A installs.

Small window, but it is #8284's exact symptom, and the veto is the one case where the user stays in the session indefinitely after having already accepted a restart, with no other signal that anything changed.

Clearing the flag in the veto branch of `quitAndInstall()` (next to `this.setState(readyState)`) closes it and keeps the "only checks for an overwrite once, so the second restart request goes through" test passing, since that test never vetoes.

**Review by jonvanausdeln — APPROVED** [review]
LGTM!

One small thing claude found that might have an effect, but seems pretty corner case, so do with that what you will 😄

File: src/vs/platform/update/electron-main/abstractUpdateService.ts (hunk 1 above)
Comment (timtmok) [reply src/vs/platform/update/electron-main/abstractUpdateService.ts:733]: Worth adding in! It seems that this is also a problem in upstream. Even though it wasn't caught there, it's good to address here.

**Comment by timtmok** [comment]
The test failure is unrelated and is already addressed in another PR.
